src/metabase/lib/join.cljc

```diff
@@ -515,28 +515,55 @@
                                         columns)]
     (concat pk fk other)))
 
+;;; this logic is different from what's in [[metabase.lib.remove-replace/remove-join]] so it is not duplicate code.
+(defn- remove-join-and-all-subsequent-joins
+  "Remove a `join-to-remove` and all subsequent joins from a query stage."
+  [query stage-number {join-to-remove-alias :alias, :as _join-to-remove}]
+  (letfn [(remove-join? [a-join]
+            (= (:alias a-join) join-to-remove-alias))
+          (update-joins [existing-joins]
+            (not-empty
+             (reduce
+              (fn [acc a-join]
+                (if (remove-join? a-join)
+                  (reduced acc)
+                  (conj acc a-join)))
+              []
+              existing-joins)))
+          (update-stage [stage]
+            (let [new-joins (update-joins (:joins stage))]
+              (u/assoc-dissoc stage :joins new-joins)))]
+    (lib.util/update-query-stage query stage-number update-stage)))
```

Comment:
I would prefer to inline this or make it clear in the name that it can produce broken queries (orphaned join field references) that should not escape to userland.

Reply:
Good point, I reworked this to do post-filtering so now we don't have an invalid query at any point